### PR TITLE
feat: add docs around jx concepts

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

# Description

First pass at adding documentation for jx concepts.
Does not talk about lot of things, but it's a start.
We can add more things subsequently.

This is also part of the 30 day plan for the GSoC interns.

Fixes # (issue)

# Checklist:

- [ ] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
- [ ] I have signed off the commit, as per instructions mentioned [here](https://jenkins-x.io/community/code/#how-to-sign-your-commits).
- [ ] Any dependent changes have already been merged.

### CODE REVIEW

File: content/en/v3/about/concepts/pipeline-activity.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Each type of step has a name, status, start and end timestamp.
+A pipeline activity can have steps of different kinds (for example a pipeline activity can have a stage, preview and promote step)
+Let's look at each of these kinds in detail.
+
+### Stage
+
+A stage step is made up of multiple sub steps and is normally associated with the Continuous Integration (CI) part of Jenkins X.
+The integration of Jenkins X with tekton is explained [here](../tekton-integration)
+
```
Comment (rajatgupta24) [inline content/en/v3/about/concepts/pipeline-activity.md:29]: I read it but I think an example would be more helpful, something like this is a stage in pa that is generated from this pr.yaml.

**Comment by ankitm123** [comment]
Seeing this issue with preview:
```
time="2022-07-01T20:37:21Z" level=error msg="googleapi: Error 404: The 'entity.change.deletions[7]' resource named 'a-jx-docs-jx-jenkins-x-jx-docs-pr-3606.infra.jenkins-x.rocks. (TXT)' does not exist.\nMore details:\nReason: notFound, Message: The 'entity.change.deletions[7]' resource named 'a-jx-docs-jx-jenkins-x-jx-docs-pr-3606.infra.jenkins-x.rocks. (TXT)' does not exist.\nReason: notFound, Message: The 'entity.change.deletions[8]' resource named 'a-jx-docs-jx-jenkins-x-jx-docs-pr-3588.infra.jenkins-x.rocks. (TXT)' does not exist.\n"
```

File: content/en/v3/about/concepts/pipeline-activity.md (hunk 1 above)
Comment (ankitm123) [reply content/en/v3/about/concepts/pipeline-activity.md:29]: Included an example in the deployment lifecycle section.

**Pull request merged**